src/checks/y_check_db_access_in_ut.clas.abap

```diff
@@ -51,131 +86,185 @@ CLASS y_check_db_access_in_ut IMPLEMENTATION.
     settings-apply_on_test_code = abap_true.
     settings-documentation = |{ c_docs_path-checks }db-access-in-ut.md|.
 
-    relevant_statement_types = VALUE #( ( scan_struc_stmnt_type-class_implementation ) ).
+    relevant_statement_types = VALUE #( ( scan_struc_stmnt_type-class_definition )
+                                        ( scan_struc_stmnt_type-class_implementation ) ).
     relevant_structure_types = VALUE #( ).
 
     set_check_message( 'Database access(es) within a Unit-Test should be removed!' ).
   ENDMETHOD.
 
 
-  METHOD inspect_statements.
-    inspect_class_definition( structure ).
+  METHOD inspect_tokens.
+    CASE structure-stmnt_type.
+      WHEN scan_struc_stmnt_type-class_definition.
+        add_line_to_defined_classes( statement = statement
+                                     structure = structure ).
+
+      WHEN scan_struc_stmnt_type-class_implementation.
+        check_class( index = index
+                     statement = statement
+                     structure = structure ).
+
+    ENDCASE.
+  ENDMETHOD.
+
+
+  METHOD is_persistent_object.
+    cl_abap_structdescr=>describe_by_name( EXPORTING p_name = obj_name
+                                           EXCEPTIONS OTHERS = 1 ).
```

Comment:
as it 'complies' the object, should we replace it with a single select in TADIR?

Reply:
i take a look